### PR TITLE
Attempt to fetch Derivative.absolute_location in WaveformJob

### PR DESCRIPTION
**Pull request by phuongdh**
To avoid pulling from the streaming server if possible.

### CODE REVIEW

File: app/jobs/waveform_job.rb [hunk 1]
```diff
@@ -51,6 +48,22 @@ def file_uri(master_file)
       end
     end
 
+    def derivative_file_uri(master_file)
+      derivatives = master_file.derivatives
+      uri = nil
+
+      # Find the lowest quality stream
+      ['high', 'medium', 'low'].each do |quality|
+        d = derivatives.select { |d| d.quality == quality }.first
```
Comment (cjcolvar) [inline app/jobs/waveform_job.rb:57]: You should be able to simplify this with `find`:
```suggestion
        d = derivatives.find { |derivative| derivative.quality == quality }
```

Comment (phuongdh) [reply app/jobs/waveform_job.rb:57]: Thanks!

Comment (phuongdh) [reply app/jobs/waveform_job.rb]: Nvm, that didn't work, I remember starting out with this now. 
```
Failure/Error: d = derivatives.find { |derivative| derivative.quality == quality }
     ArgumentError:
       Couldn't find Derivative without an ID
```

Comment (cjcolvar) [reply app/jobs/waveform_job.rb]: Ah, probably needs to be `derivatives.to_a.find` due to AF.

**Pull request merged**